tests/internal/e2e/test/specs/vcast_coded_test_completion.test.ts

```diff
@@ -0,0 +1,416 @@
+    }
+
+    if (!subprogram) {
+      throw "Subprogram 'manager' not found";
+    }
+
+    console.log("Looking for coded tests");
+    const subprogramMethod = await findSubprogramMethod(
+      subprogram,
+      "Coded Tests"
+    );
+    if (!subprogramMethod) {
+      throw "Subprogram method 'Coded Tests' not found";
+    }
+
+    if (!subprogramMethod.isExpanded()) {
+      await subprogramMethod.select();
+    }
+
+    let contextMenu = await subprogramMethod.openContextMenu();
+    console.log("Generating template test");
+    await contextMenu.select("VectorCAST");
+    let menuElement = await $("aria/Generate New Coded Test File");
+    await menuElement.click();
+
+    await (await $("aria/Save Code Test File")).click();
+    for (const character of "TestFiles/manager-template.cpp") {
+      await browser.keys(character);
+    }
+
+    await browser.keys(Key.Enter);
+
+    await bottomBar.openOutputView();
+    console.log("Checking that tests got generated");
+    let testHandle = await getTestHandle(
+      subprogram,
+      "Coded Tests",
+      "managerTests.ExampleFixtureTestCase",
+      2
+    );
+    expect(testHandle).not.toBe(undefined);
+
+    testHandle = await getTestHandle(
+      subprogram,
+      "Coded Tests",
+      "managerTests.ExampleTestCase",
+      2
+    );
+    expect(testHandle).not.toBe(undefined);
+
+    contextMenu = await testHandle.openContextMenu();
+    await contextMenu.select("VectorCAST");
+    menuElement = await $("aria/Edit Coded Test");
+    await menuElement.click();
+
+    const editorView = workbench.getEditorView();
+    await browser.waitUntil(
+      async () =>
+        (await (await editorView.getActiveTab()).getTitle()) ===
+        "manager-template.cpp"
+    );
+
+    // Insert "// vmock" on line 16
+    const tab = (await editorView.openEditor(
+      "manager-template.cpp"
+    )) as TextEditor;
+    await browser.keys([Key.Ctrl, Key.Space]);
+    const contentAssist = await tab.toggleContentAssist(true);
+    await tab.setTextAtLine(14, "// vmock");
+
+    // Ensure cursor is positioned after "// vmock"
+    let currentLine = await tab.getLineOfText("// vmock");
+    await tab.typeTextAt(currentLine, "// vmock".length + 1, " ");
+    await tab.save();
+    await browser.waitUntil(
+      async () => (await contentAssist.getItems()).length > 0
+    );
+
+    // Validate content assist items
+    console.log("Validating content assist for '// vmock'");
+    expect(await contentAssist.hasItem("database")).toBe(true);
+    expect(await contentAssist.hasItem("manager")).toBe(true);
+    expect(await contentAssist.hasItem("Prototype-Stubs")).toBe(true);
+
+    await selectItem(contentAssist, "manager");
+    await tab.typeTextAt(currentLine, "// vmock manager".length + 1, " ");
+    await tab.save();
+    await browser.waitUntil(
+      async () => (await contentAssist.getItems()).length > 0
+    );
+
+    expect(await contentAssist.hasItem("Manager::AddIncludedDessert")).toBe(
```

Comment:
Since there are a lot of assertions, you could iterate over a list of expected items

Reply:
Refactored in b797c7864c92738222537ffd5e50d5fc216835ba .